includes/Core/Conversion_Tracking/Conversion_Event_Providers/WooCommerce.php

```diff
@@ -271,7 +277,7 @@ function () {
 	 *
 	 * @return array
 	 */
-	protected function get_formatted_product( WC_Product $product, $variation_id = 0, $variation = false, $quantity = false ) {
```

Comment:
I don't think this is the problem unless we're not guarding the way its called properly (which at least at this point, we are). The main problem is that this class isn't imported (or using fully-qualified path) which is indicated by the error reported in QA:
```
($product) must be of type Google\Site_Kit\Core\Conversion_Tracking\Conversion_Event_Providers\WC_Product
```

The simplest solution is to import all references (which is safe even if they don't exist).

Reply:
Yes. Instance of will match any sub class so that's fine.

As for the import it won't fatal until the class is used, it's just defining the shortcut to the fully qualified class name in the file.